src/NuGet.Core/NuGet.Protocol.Core.v2/DownloadResourceV2.cs

```diff
@@ -40,6 +40,11 @@ public DownloadResourceV2(V2Resource resource)
                 throw new ArgumentNullException(nameof(identity));
             }
 
+            if (logger == null)
+            {
+                logger = new NuGet.Logging.NullLogger();
```

Comment:
I am not sure about this. If the caller does not have a logger, like test code, they can pass in a NullLogger. If logger is null in this method, we should throw ArgNull